Source/NSSwitch.m

```diff
@@ -0,0 +1,259 @@
+/* Implementation of class NSSwitch
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: Wed Apr  8 22:01:02 EDT 2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "AppKit/NSSwitch.h"
+#import "GNUstepGUI/GSTheme.h"
+
+@implementation NSSwitch
+
++ (void) initialize
+{
+  if (self == [NSSwitch class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) setState: (NSControlStateValue)s
+{
+  _state = s;
+  [self setNeedsDisplay];
+}
+
+- (NSControlStateValue) state
+{
+  return _state;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setTarget: (id)target
+{
+  _target = target;
+}
+
+- (id) target
+{
+  return _target;
+}
+
+- (void) setDoubleValue: (double)val
+{
+  if (val < 1.0)
+    {
+      [self setState: NSControlStateValueOff];
+    }
+  else
+    {
+      [self setState: NSControlStateValueOn];
+    }
+}
+
+- (double) doubleValue
+{
+  return (double)(([self state] == NSControlStateValueOn) ? 1.0 : 0.0);
+}
+
+- (void) setFloatValue: (float)val
+{
+  [self setDoubleValue: (double)val];
+}
+
+- (float) floatValue
+{
+  return (float)[self doubleValue];
+}
+
+- (void) setIntValue: (int)val
+{
+  [self setDoubleValue: (double)val];
+}
+
+- (int) intValue
+{
+  return (int)[self doubleValue];
+}
+
+- (void) setIntegerValue: (NSInteger)val
+{
+  [self setDoubleValue: (double)val];
+}
+
+- (NSInteger) integerValue
+{
+  return (NSInteger)[self doubleValue];
+}
+
+- (void) setStringValue: (NSString *)val
+{
+  [self setDoubleValue: [val doubleValue]];
+}
+
+- (NSString *) stringValue
```

Comment:
What about objectValue?
And does an NSSwitch have a tag and what other NSControl methods need to be implemented to make this class usable?
Are there bindings?

Reply:
There are no bindings.

Reply:
Yes, it does have a tag.  It gets this from its parent NSView.

Reply:
I believe I have implemented everything that it needs now to be useful.